presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -495,6 +521,51 @@ private static RuntimeException propagatePrestoException(Throwable throwable)
         void finish();
 
         boolean isFinished(OptionalInt bucketNumber);
+
+        // returns the number of finished splits being rewinded
```

Comment:
Sure. But we still need to specify "finished" here because the number only reflects rewinded finished InternalHiveSplits so that the tracking is correct.